docs/data/date-pickers/lifecycle/lifecycle.md

```diff
@@ -0,0 +1,333 @@
+even if the other date does not.
+
+In the example below, changing the value of the start date section will call `onChange` even if the end date is empty or partially filled.
+
+{{"demo": "LifeCycleDateRangeField.js", "defaultCodeOpen": false}}
+
+## Pickers lifecycle
+
+### When is `onClose` called ?
+
+:::info
+If you are not controlling the `open` prop, then all the scenarios below also describe when the picker closes.
+:::
+
+#### When the last view is completed
+
+When the last view is completed, `onClose` will be called only if the `closeOnSelect` prop is equal to `true`.
+By default, it is set to `true` on desktop and `false` on mobile.
+
+Here are a few examples:
+
+:::info
+The example below are using the desktop and mobile variant of the pickers, but the behavior is exactly the same when using the responsive variant (`DatePicker`, `TimePicker`, ...) on a mobile or desktop environment.
+:::
+
+- ```tsx
+  <DesktopDatePicker />
+  ```
+
+  - Default `views` prop: `['year', 'month', 'day']`
+  - Default `closeOnSelect` prop: `true`
+
+  **Behavior:** The picker will close when selecting the day.
+
+- ```tsx
+  <DesktopDatePicker closeOnSelect={false} />
+  ```
+
+  - Default `views` prop: `['year', 'month', 'day']`
+  - Explicit `closeOnSelect` prop: `false`
+
+  **Behavior:** The picker won't close when selecting a value.
+
+  :::warning
+  If you want to set `closeOnSelect` to `false` on a desktop picker, you should probably also enable the action bar to allow the user to validate the value:
+
+  ```tsx
+  <DesktopDatePicker
+    closeOnSelect={false}
+    slotProps={{ actionBar: { actions: ['cancel', 'accept'] } }}
+  />
+  ```
+
+  :::
+
+- ```tsx
+  <DesktopDatePicker views={['day', 'month', 'year']} />
+  ```
+
+  - Explicit `views` prop: `['day', 'month', 'year']`
+  - Default `closeOnSelect` prop: `true`
+
+  **Behavior:** The picker will close when selecting the year
+
+- ```tsx
+  <DesktopTimePicker />
+  ```
+
+  - Default `views` prop: `['hours', 'minutes']` (plus a `meridiem` view because the default local is in 12-hours format)
+  - Default `closeOnSelect` prop: `true`
+
+  **Behavior:** The picker will close when selecting the meridiem
+
+:::info
+You don't have to fill all the views for the picker to close automatically.
+For example, on the `DatePicker`, the `year` and `month` views are not in the default workflow since they are before the opening view (`day`),
+so the picker will close even if you never went to those views.
+:::
+
+#### When the picker is manually closed
+
+Pressing <kbd class="key">Escape</kbd> or clicking outside the picker will close the picker.
+
+#### When a value is selected using the action bar
+
+Clicking on any built-in button of the action bar will close the picker.
+
+#### When a shortcut is picked
+
+Clicking on a shortcut will close the picker, except if the `changeImportance` property has been set to `"set"` instead of `"accept"`.
```

Comment:
```suggestion
Clicking on a shortcut will close the picker, except if the `changeImportance` property has been set to `"set"` instead of the default value `"accept"`.
```